Gemfile

```diff
@@ -164,6 +164,10 @@ gem 'oj_mimic_json'
 # CORS for local testing/dev
 gem 'rack-cors'
 
+# Salesforce streaming API
+gem 'cookiejar', git: 'https://github.com/MissionCapital/cookiejar.git'
+gem 'faye', '0.8.9'
```

Comment:
The version of cookiejar (latest 0.3.3 2016) and faye (latest 1.4.0 2020) are not the latest. Cookiejar is a fork of the original that is behind in commits from the original. Why these versions?

Reply:
Good catch!
This is the blog post I read to get started with this (written in 2020). https://medium.com/@sahsantoshh/salesforce-integration-with-ruby-on-rails-for-real-time-changes-ff93dde05ba4

I tried to update faye to the latest version but the stream connection kept dropping every 10-20 minutes. There was a github issue that I can't find now that mentioned having to use faye 0.8.9 with the salesforce streaming API. 
So, I don't know - but I do know this version is required for it to work because the newer one didn't.

I'll checkout the cookiejar, which might be okay to use from the original package.

Reply:
Actually, the version on rubygems of cookiejar (0.3.3 from 2016) is nearly a year older than the version being used here https://github.com/MissionCapital/cookiejar/commit/68eecc8415338c6e9d636c4f7e4d2e13547e20a2 - from 2017).

Reply:
Sorry I missed that.